xbmc/guilib/GUIEditControl.cpp

```diff
@@ -24,38 +24,58 @@
 #include "utils/Variant.h"
 #include "windowing/WinSystem.h"
 
+#include <algorithm>
+
 using namespace KODI::GUILIB;
 
 using KODI::UTILITY::CDigest;
 
-const char* CGUIEditControl::smsLetters[10] = { " !@#$%^&*()[]{}<>/\\|0", ".,;:\'\"-+_=?`~1", "abc2ABC", "def3DEF", "ghi4GHI", "jkl5JKL", "mno6MNO", "pqrs7PQRS", "tuv8TUV", "wxyz9WXYZ" };
-const unsigned int CGUIEditControl::smsDelay = 1000;
-
 #ifdef TARGET_WINDOWS
 extern HWND g_hWnd;
 #endif
 
+namespace
+{
+constexpr std::string_view smsLetters[] = {" !@#$%^&*()[]{}<>/\\|0",
+                                           ".,;:\'\"-+_=?`~1",
+                                           "abc2ABC",
+                                           "def3DEF",
+                                           "ghi4GHI",
+                                           "jkl5JKL",
+                                           "mno6MNO",
+                                           "pqrs7PQRS",
+                                           "tuv8TUV",
+                                           "wxyz9WXYZ"};
+
+constexpr float smsDelay = 1000;
+
+// Additional space between left label text and left label text
+constexpr float TEXT_SPACE = 20.0f;
```

Comment:
What are the units here? Does this assume a 1920x1080 skin?

Reply:
the use of "space" variables along most gui controls has been done more than 10y old
and the only ancient refactor do not explains nothing,
for sure are pixels, i too have theorised for 1920x1080 i believe no documentation exists

Reply:
for example in a 1080p display, if you put kodi windowed, by debugging all control rendering variables
you still have values for full 1080p, seem all them are scaled to another place

Reply:
I think I understand that. You mean a skin that defines a 1080p folder, right? if you put Kodi windows, all control dimensions are scaled to 1080p. But what about a 720p skin like Confluence? Will that still scale by 720 or will it scale by 1080?

Reply:
these GUI thing has always confused me because are not so clear,
what i can say is that properties like gui control width/height has px based on the xml skin resolution
but other values like x/y are referred to current display resolution widowed or full
but by debugging ProcessText/RenderText i remember that if i am in windowed mode i have to take in account that calculated coordinates values are based full screen, i cannot tell you specifically how the kodi GUI works because it would take days to try investigate
but in any case these are pixel unit

Reply:
but whats your request on this variable?

Reply:
Chose one of two requests:

* Spend days, if not years, figuring out what the heck is going on
* Do what works, if this variable works for you, do it

I'd choose the second :wink:

Reply:
yeah i also choose the second one😐
this variable had a different name before just renamed and increased value other gui controls do the same so good as is

Reply:
rebased and add "in pixels" in the description